### PR TITLE
OCPBUGS-8711: API-1509: Enable AES-GCM encryption

### PR DESCRIPTION
**Pull request by dgrisonnet**
This PR is based on https://github.com/openshift/library-go/pull/1469 and https://github.com/openshift/library-go/pull/1475 to enable aesgcm encryption and add tests verifying that the feature works.

It splits the existing encryption e2e tests per encryption provider and add the relevant new Makefile rules to make sure that we have dedicated CI jobs for each provider. This is meant to reduce the duration of the tests.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -36,29 +38,50 @@ $(call add-crd-gen,manifests,$(CRD_APIS),./manifests,./manifests)
...
 $(call verify-golang-versions,Dockerfile.rhel7)
 
+TEST_E2E_ENCRYPTION_TARGETS=$(addprefix test-e2e-encryption-,$(ENCRYPTION_PROVIDERS))
+
 # these are extremely slow serial e2e encryption tests that modify the cluster's global state
 test-e2e-encryption: GO_TEST_PACKAGES :=./test/e2e-encryption/...
 test-e2e-encryption: GO_TEST_FLAGS += -v
 test-e2e-encryption: GO_TEST_FLAGS += -timeout 4h
 test-e2e-encryption: GO_TEST_FLAGS += -p 1
+test-e2e-encryption: GO_TEST_ARGS += -args -provider=$(ENCRYPTION_PROVIDER)
```
Comment (dgrisonnet) [inline Makefile:48]: This is not yet supported by https://github.com/openshift/build-machinery-go, I will send a PR to add it.

Comment (dgrisonnet) [reply Makefile:48]: https://github.com/openshift/build-machinery-go/pull/74?notification_referrer_id=NT_kwDOAVSxGbM1NzMyMjI2NTg4OjIyMzI3NTc3#event-8632373621

**Comment by dgrisonnet** [comment]
Retesting now that aesgcm jobs are running in TechPreviewNoUpgrade clusters.

**Comment by mfojtik** [comment]
blocker bug for disabled test exists

/lgtm
/hold

(unhold if you are ready to merge)

**Comment by dgrisonnet** [comment]
The `e2e-gcp-operator-encryption-perf-single-node` failure is valid but it is caused by the fact that `TestPerfEncryption` is not resilient to disruption. I observed that flake a few weeks ago, and I opened https://issues.redhat.com/browse/API-1553 to fix it later. Since it is only a test issue, it is fine to move forward and disregard this failure for now.

**Pull request merged**